.github/workflows/intergration-test.yml

```diff
@@ -0,0 +1,36 @@
+name: "Integration Test"
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  integration_test:
+    name: "Integration Test"
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go: [ "1.15", "1.16" ]
+        ipfs: ["0.8", "0.9"]
+
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up IPFS ${{ matrix.ipfs }}
+        uses: ibnesayeed/setup-ipfs@master
+        id: ipfs_setup
+        with:
+          ipfs_version: ${{ matrix.ipfs }}
+          run_daemon: true
+
+      - name: Test
+        env:
+          STORAGE_IPFS_INTEGRATION_TEST: ${{ secrets.STORAGE_INTEGRATION_TEST }}
+          STORAGE_IPFS_NAME: ${{ secrets.STORAGE_IPFS_NAME }}
+          STORAGE_IPFS_ENDPOINT: ${{ secrets.STORAGE_IPFS_ENDPOINT }}
+        run: make integration_test
```

Comment:
What's your idea about the integration test in CI? 

For cloud services like s3, we have to use a test account. But for IPFS, we can start the IPFS node in either a separate server or just in the GitHub action runner (maybe with https://docs.github.com/en/actions/guides/about-service-containers)? What do you think about it?

Reply:
If you want to think more about it, we can also remove the GH action yml and add it in another PR.

Reply:
I am now running the IPFS service directly in the GitHub Actions runner, should I switch to service containers, or keep both?

Reply:
I'm sorry that I missed your work here. Let me take a look.